### ISSUE TITLE
Inconsistent return type when using Promise.resolve.

### ISSUE THREAD

**Issue by TheModMaker** [google/closure-compiler#1887]
When using `Promise.resolve` and record types, I get an inconsistent return type error even though the types are exactly the same.  This only appears to happen with record types and when using `Promise.resolve`, when using chained Promises, it correctly determines the type.

``` javascript
/** @return {!Promise<{uri: string}>} */
function Test() {
  return Promise.resolve({uri: 'abc'});
}
```

```
input0:3: WARNING - inconsistent return type
found   : Promise<{uri: string}>
required: Promise<{uri: string}>
  return Promise.resolve({uri: 'abc'});
         ^

0 error(s), 1 warning(s), 100.0% typed
```

https://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540return%2520%257B!Promise%253C%257Buri%253A%2520string%257D%253E%257D%2520*%252F%250Afunction%2520Test()%2520%257B%250A%2520%2520return%2520Promise.resolve(%257Buri%253A%2520'abc'%257D)%253B%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by Dominator008** [comment]
Seems to work in NTI.

**Comment by lauraharker** [comment]
The repro doesn't warn now, seems to have been fixed at some point. Closing the issue.
